base/broadcast.jl

```diff
@@ -448,8 +442,18 @@ Note that `dest` is only used to store the result, and does not supply
 arguments to `f` unless it is also listed in the `As`,
 as in `broadcast!(f, A, A, B)` to perform `A[:] = broadcast(f, A, B)`.
 """
-@inline broadcast!(f, C::AbstractArray, A, Bs::Vararg{Any,N}) where {N} =
-    _broadcast!(f, C, A, Bs...)
+broadcast!(f, dest, As...) = broadcast!(f, dest, combine_styles(As...), As...)
+broadcast!(f, dest, ::BroadcastStyle, As...) = broadcast!(f, dest, nothing, As...)
+@inline function broadcast!(f, C, ::Void, A, Bs::Vararg{Any,N}) where N
```

Comment:
I didn't think about this before, but I wonder if we should consider the implications of specializing  `A, Bs::Vararg{Any,N}` vs just `As::Vararg{Any,N}`. I believe those two have separate precedence for dispatch. Whatever we choose should be documented that way.